tests/hsm-test.rs

```diff
@@ -0,0 +1,240 @@
+// SPDX-FileCopyrightText: Wiktor Kwapisiewicz <wiktor@metacode.biz>
+// SPDX-FileCopyrightText: Heiko Schaefer <heiko@schaefer.name>
+// SPDX-License-Identifier: Apache-2.0 OR MIT
```

Comment:
please no license headers, if you want to keep a reference to the original authors feel free to leave a regular comment

Reply:
good point. that's unintentional copypasta.